### PR TITLE
Improve Bigint

### PR DESCRIPTION
**Pull request by juancarlospaco**
- https://github.com/timotheecour/Nim/issues/566#issue-799927295

### CODE REVIEW

File: lib/std/jsbigints.nim [hunk 1]
```diff
@@ -8,12 +8,14 @@ type JsBigInt* = distinct JsBigIntImpl        ## Arbitrary precision integer for
 
 func big*(integer: SomeInteger): JsBigInt {.importjs: "BigInt(#)".} =
   ## Constructor for `JsBigInt`.
+  when nimvm: doAssert off, "JsBigInt can not be used at compile-time nor static context" else: discard
```
Comment (ringabout) [inline lib/std/jsbigints.nim]:
```suggestion
  when nimvm: doAssert false, "JsBigInt can not be used at compile-time nor static context" else: discard
```

File: lib/std/jsbigints.nim [hunk 2]
```diff
@@ -8,12 +8,14 @@ type JsBigInt* = distinct JsBigIntImpl        ## Arbitrary precision integer for
...
   runnableExamples:
     doAssert big(1234567890) == big"1234567890"
     doAssert 0b1111100111.big == 0o1747.big and 0o1747.big == 999.big
 
 func big*(integer: cstring): JsBigInt {.importjs: "BigInt(#)".} =
   ## Constructor for `JsBigInt`.
+  when nimvm: doAssert off, "JsBigInt can not be used at compile-time nor static context" else: discard
```
Comment (ringabout) [inline lib/std/jsbigints.nim]:
```suggestion
  when nimvm: doAssert false, "JsBigInt can not be used at compile-time nor static context" else: discard
```

**Review by ringabout — APPROVED** [review]

File: lib/std/jsbigints.nim [hunk 3]
```diff
@@ -8,12 +8,14 @@ type JsBigInt* = distinct JsBigIntImpl        ## Arbitrary precision integer for
 
 func big*(integer: SomeInteger): JsBigInt {.importjs: "BigInt(#)".} =
   ## Constructor for `JsBigInt`.
+  when nimvm: doAssert false, "JsBigInt can not be used at compile-time nor static context" else: discard
```
Comment (Araq) [inline lib/std/jsbigints.nim:11]: Can the `error` pragma be used here instead?

File: lib/std/jsbigints.nim [hunk 4]
```diff
@@ -8,12 +8,14 @@ type JsBigInt* = distinct JsBigIntImpl        ## Arbitrary precision integer for
...
   runnableExamples:
     doAssert big(1234567890) == big"1234567890"
     doAssert 0b1111100111.big == 0o1747.big and 0o1747.big == 999.big
 
 func big*(integer: cstring): JsBigInt {.importjs: "BigInt(#)".} =
   ## Constructor for `JsBigInt`.
+  when nimvm: doAssert false, "JsBigInt can not be used at compile-time nor static context" else: discard
```
Comment (Araq) [inline lib/std/jsbigints.nim:18]: Same question.

File: lib/std/jsbigints.nim (hunk 3 above)
Comment (juancarlospaco) [reply lib/std/jsbigints.nim:11]: Nope, because it errors out both ways, even with code that should compile, I tried. Good observation anyway.

Comment (timotheecour) [reply lib/std/jsbigints.nim:11]: same as https://github.com/nim-lang/Nim/pull/17161#discussion_r583124021

**Pull request merged**